tests/chunked_worker.py

```diff
@@ -0,0 +1,68 @@
+import collections
+import typing
+import uuid
+
+from dss.events.chunkedtask import Runner, Runtime, Task
+
+
+def task_lifecycle(
+        initial_task: typing.Type[Task],
+        initial_state: typing.Any,
+        runtime_builder: typing.Callable[[typing.Mapping[str, typing.Any]], Runtime],
+        task_builder: typing.Callable[[typing.Type[Task], typing.Any, Runtime], Task],
+        runtime_result: typing.Callable[[Runtime], typing.Optional[typing.Any]],
+        runtime_scheduled_getter: typing.Callable[
+            [Runtime], typing.List[typing.Tuple[typing.Type[Task], typing.Any, typing.Optional[str]]]],
+) -> typing.Tuple[int, typing.Any]:
+    """
+    Run through the lifecycle of a task all the way to completion.  The initial task's class and state is passed into
+    the method, and various callables must be provided to complete the task.
+
+    :param initial_task: This is the initial task that we're starting with.
+    :param initial_state: This is the initial state of the task we're starting with.
+    :param runtime_builder: This is a callable that is called to generate a Runtime object.  It is called with a
+                            dictionary that stores a mapping between task uuids and their results.
+    :param task_builder: This is a callable that produces a Task object, given a task's class, the state, and the
+                         runtime.
+    :param runtime_result: This is a callable that we execute against a runtime object to get the result of the task.
+                           If the task is not complete, the callable should return None.
```

Comment:
In addition to the comment in the other PR about `task_lifecycle`, consider also making `runtime_result` something more action-y. `get_runtime_result`?

Reply:
Renamed `task_lifecycle` to `run_task_to_completion`.

Renamed `runtime_result` to `runtime_result_getter` (to match `runtime_scheduled_getter`)